Automattic/pocket-casts-ios
# EffectsPlayer: change retry mechanism to stop on first failure

We've been receiving reports from users that their playback is resuming automatically after they put the AirPods on the case:

```
2023-09-28 09:30:02 EffectsPlayer: failed to start playback (1/3: Error Domain=com.apple.coreaudio.avfaudio Code=0 "(null)"
2023-09-28 09:30:02 pausing playback
2023-09-28 09:30:02 saving played up to 1787.235918367347 for episode Cormack McCarthy's Spring Break
2023-09-28 09:30:03 PlaybackManager handleAudioInterrupt began
2023-09-28 09:30:03 Loading Cormack McCarthy's Spring Break with UUID c7ed4e20-185f-486f-a47a-b5d705eaa453 autoPlay false
2023-09-28 09:30:03 saving played up to 1787.235918367347 for episode Cormack McCarthy's Spring Break
2023-09-28 09:30:03 cleanupCurrentPlayer permanent? false
2023-09-28 09:30:03 Using EffectsPlayer
2023-09-28 09:30:03 activating audio session succeeded
2023-09-28 09:30:08 Remote control: pauseCommand
```

This PR changes the retry mechanism to just pause and do nothing — fixing this error while fixing the recurring crash we used to see.

## Discussion

I'm targeting `trunk` with this change. We could create a new remote feature flag and we release that right away but I'm not sure if that's the correct approach.

Open to any ideas.

## To test

Comment those lines on `EFfectsPlayer` to force a crash:

```
strongSelf.engine?.prepare()
try strongSelf.engine?.start()
```

1. Run the app
2. Play a downloaded episode or local file
3. ✅ The app should not crash
4. Check the logs, you should see three `Loading <episode title>`, and a log like `EffectsPlayer: failed to start playback`

## Checklist
